### DISCUSSION TITLE
You might be able to use jsrepo to install components more easily

### DISCUSSION THREAD

**Discussion by ieedan** [bossadizenith/Framer-ground#10] [Ideas]
I am building a project called [jsrepo](https://jsrepo.dev) and I was noticing the process to install components is 100% manual.

I think you could automate this for your users pretty easily.

I was just trying this out on my fork https://github.com/ieedan/Framer-ground/tree/test-jsrepo basically you could just build like:
```bash
jsrepo build --dirs ./components
```

and then in the docs instruct users to install like:
```
jsrepo add navbar/vercel
```

Might be a bit cleaner than the current process. I think it could almost be drop-in with the way you have things structured as well.

You can see what it looks like by running:
```
npx jsrepo init --repos https://github.com/ieedan/Framer-ground/tree/test-jsrepo

npx jsrepo add
```

**Comment by bossadizenith** [comment]
@ieedan that's a very cool idea.

currently I'm working on the v2 and i think it'll be a great experience for me to play around with the jsrepo.

i just watched the video on the site showcasing how it works and I must it's really good!

**Reply by ieedan** [reply]
Let me know how it works for you if you do!

**Reply by bossadizenith** [reply]
sure everything sounds good to me. will definitely give it a shot!

**Reply by bossadizenith** [reply]
@ieedan idk if you help to help me setup the cli for Ground?

**Reply by ieedan** [reply]
From what I can see things are going to need to be moved around a bit to allow jsrepo to work.

The main problem is that `./fg` has all the components but it depends on `./components` which has things that are named the same as in `./fg` so they clash. You can look at https://github.com/ieedan/shadcn-svelte-extras or https://github.com/DavidHDev/react-bits as an example of how to structure the project to make it easy to use jsrepo.

An ideal project structure would probably look like:
```
root
├── package.json
├── lib
├── hooks
├── public
└── components
    ├── fg
    │   ├── buttons
	│   └── ...
    └── ui
```

Where fg only relies on `ui`, `lib`, `hooks`, and `public`. It's also important to keep in mind that multi-level dependencies will be resolved so if `lib/utils` depends on `components/ui/...` then that also has to be included.

From my initial attempts unless the project structure changes I don't think it's possible to use jsrepo.
